odoo/models.py

```diff
@@ -2406,10 +2431,18 @@ def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
 
         data['__domain'] = expression.AND(sections)
         if len(groupby) - len(annotated_groupbys) >= 1:
-            data['__context'] = { 'group_by': groupby[len(annotated_groupbys):]}
+            data['__context'] = {'group_by': groupby[len(annotated_groupbys):]}
         del data['id']
         return data
 
+    @api.model
+    def _read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        """
+        Executes exactly what the public read_group, except it doesn't order the manyToOne on their names but on their ID instead
+        """
+        return self.with_context({'ignore_sort_x2many_name': True}).read_group(domain, fields, groupby, offset, limit,
+                                                                               orderby, lazy)
```

Comment:
```suggestion
        return self.with_context(ignore_sort_x2many_name=True).read_group(domain, fields, groupby, offset, limit,
                                                                               orderby, lazy)
```

Providing a dict to with_context replaces the whole context, which means you are losing any information provided through the context (especially the `active_test` content which modifies the returned values of `read_group`)